src/Logging/Logging.EventSource/src/EventSourceLogger.cs

```diff
@@ -190,19 +197,20 @@ private ExceptionInfo GetExceptionInfo(Exception exception)
         /// </summary>
         private IEnumerable<KeyValuePair<string, string>> GetProperties(object state)
         {
-            var arguments = new List<KeyValuePair<string, string>>();
-            var asKeyValues = state as IEnumerable<KeyValuePair<string, object>>;
-            if (asKeyValues != null)
+            if (state is IEnumerable<KeyValuePair<string, object>> keyValuePairs)
             {
```

Comment:
Is there no way to write the object directly as EventData before turning it into a list of string, string?

Reply:
I suspect there is but it's not easy. I didn't touch methods that take IEnumerable for the first pass.

Reply:
Ok we need to do that though. I’m ok looking into that next. I think it’s worrhwhile to do.